lib/elixir/lib/supervisor.ex

```diff
@@ -532,9 +532,12 @@ defmodule Supervisor do
   @doc """
   Starts a supervisor with the given children.
 
-  The children is a list of modules, two-element tuples with module and
-  arguments or a map with the child specification. A strategy is required
-  to be provided through the `:strategy` option. See
+  `children` is a list that includes every child expressed in of the following forms:
+    - a module
+    - a two-element tuple in the shape of `{module, arguments}` with `arguments` being a keyword-list
+    - a [child specification](`t:child_spec/0`)
```

Comment:
```suggestion
  `children` is a list of the following forms:

    * a module where `module.child_spec([])` will be invoked to retrieve
      its child specification

    * a two-element tuple in the shape of `{module, arg}` where `module.child_spec(arg)`
      will be invoked for retrieve its child specification

    * a [child specification](`t:child_spec/0`)
```

Reply:
The idea is of the PR is to make it clear that a list combining these elements is valid, unless it is not and I'm wrong.

Reply:
I am not sure I read the proposal above as saying it must be "only one of those". Lists in Elixir are heterogeneous, so that is the default assumption in all APIs, unless otherwise state?

Reply:
You are right.